tools/content/fetch_messages.dart

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env dart
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+// Avoid any Flutter-related dependencies so this can be run as a CLI program.
+import 'package:args/args.dart';
+import 'package:http/http.dart';
+import 'package:ini/ini.dart' as ini;
+import 'package:zulip/api/backoff.dart';
+
+import 'model.dart';
+
+/// Fetch all public message contents from a Zulip server in bulk.
+///
+/// It outputs JSON entries of the message IDs and the rendered HTML contents in
+/// JSON Lines (https://jsonlines.org) format. The output can be used later to
+/// perform checks for discovering unimplemented features.
+///
+/// Because message IDs are only unique within a single server, the script
+/// names corpuses from each server differently (if --corpus-dir is specified).
+///
+/// This script is meant to be run via `tools/content/check-features`.
+///
+/// For more help, run `tools/content/fetch_message.dart --help`.
+///
+/// See also:
+/// * tools/content/unimplemented_features_test.dart, which runs checks against
+///   the fetched corpuses.
+void main(List<String> args) async {
+  final argParser = ArgParser();
+  argParser.addOption(
+    'config-file',
+    help: 'A zuliprc file with identity information including email, API key\n'
+          'and the Zulip server URL to fetch the messages from (required).\n\n'
+          'To get the file, see\n'
+          'https://zulip.com/api/configuring-python-bindings#download-a-zuliprc-file.',
+    valueHelp: 'path/to/zuliprc',
+  );
+  argParser.addOption(
+    'corpus-dir',
+    help: 'The directory to look for/store the corpus file.  If not given,\n'
+          'the script will write output to stdout.  Otherwise, this will\n'
+          'first read from the existing corpus file (assumed to be named\n'
+          'as "host name of the Zulip server.jsonl") to avoid duplicates\n'
+          'before fetching more messages',
+    valueHelp: 'path/to/czo.jsonl',
```

Comment:
This example value looks like it doesn't match the meaning — it looks like it's meant as the name of the file itself, rather than a directory.